README.md

```diff
@@ -1,207 +1,237 @@
-variables.**
-
-**Note: To enable debug logging for troubleshooting assistance, please set the
+| `signalfx.trace.annotated.method.blacklist` | `SIGNALFX_TRACE_ANNOTATED_METHOD_BLACKLIST` | `null` | Prevents `@Trace` annotation functionality for the target method string of format `package.OuterClass[methodOne,methodTwo];other.package.OuterClass$InnerClass[*];`. All methods in class need to include `*` and `;`. |
+| `signalfx.max.spans.per.trace` | `SIGNALFX_MAX_SPANS_PER_TRACE` | `0 (no limit)` | Drops traces with more spans than this. Intended to prevent runaway traces from flooding upstream systems. |
+| `signalfx.max.continuation.depth` | `SIGNALFX_MAX_CONTINUATION_DEPTH` | `100` | Stops propagating asynchronous context at this recursive depth. Intended to prevent runaway traces from leaking memory. |
+
+### Steps
+
+To send traces to a local or remote Smart Agent or OpenTelemetry Collector, specify
+the `signalfx.agent.host` system property or `SIGNALFX_AGENT_HOST` environment
+variable before you include the agent in your application. If you send traces
+to a Smart Agent, specify a `v1` endpoint. If you send traces to an
+OpenTelemetry Collector, specify a `v2` endpoint.
+
+To send traces to a SignalFx ingest endpoint, specify the `signalfx.endpoint.url`
+system property or `SIGNALFX_ENDPOINT_URL` environment variable before you
+include the agent in your application. If you're using µAPM PM, specify a `v1`
+endpoint in the URL. If you're using the latest µAPM, specify a `v2` endpoint
+in the URL.
+
+Follow these steps to configure the agent to send traces for `your_app` to a
+local Smart Agent available on `localhost`.
+
+1. Download the latest version of the agent from the
+[releases](https://github.com/signalfx/signalfx-java-tracing/releases) page.
+2. Set the required environment variables or system properties for your
+application. For more information about the required environment variables,
+see the [Configuration values](#configuration-values).
+This is how you can set the proper environment variables from the command line:
+    ```bash
+    $ export SIGNALFX_SERVICE_NAME="your_app"
+    $ export SIGNALFX_AGENT_HOST="localhost"
+    $ export SIGNALFX_TRACING_ENABLED="true"
+    $ export SIGNALFX_SPAN_TAGS="key1:val1,key2:val2"
+    $ export SIGNALFX_DB_STATEMENT_MAX_LENGTH=1024
+    $ export SIGNALFX_RECORDED_VALUE_MAX_LENGTH=12288
+    $ export SIGNALFX_TRACE_ANNOTATED_METHOD_BLACKLIST=null
+    $ export SIGNALFX_MAX_SPANS_PER_TRACE=0
+    $ export SIGNALFX_MAX_CONTINUATION_DEPTH=100
+    ```
+1. Include the agent in your Java application: 
+    ```bash
+    $ java -javaagent:path/to/signalfx-tracing.jar -jar app.jar
+    ```
+
+## Troubleshoot the SignalFx Java Agent
+
+Enable debug logging for troubleshooting assistance. Set the
 `datadog.slf4j.simpleLogger.defaultLogLevel=debug` property at runtime.  These logs
-are extremely verbose.**
+are extremely verbose.
```

Comment:
Should have a note about how this should only be enabled when needed and immediately turned off -- it will impact performance on the application.